test/e2e/csv_e2e_test.go

```diff
@@ -39,15 +40,351 @@ var _ = Describe("ClusterServiceVersion", func() {
+							Type:      v1alpha1.InstallModeTypeAllNamespaces,
+							Supported: true,
+						},
+					},
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), &csv)).To(Succeed())
+
+			Eventually(func() ([]v1alpha1.RequirementStatus, error) {
+				if err := ctx.Ctx().Client().Get(context.Background(), client.ObjectKeyFromObject(&csv), &csv); err != nil {
+					return nil, err
+				}
+				var result []v1alpha1.RequirementStatus
+				for _, s := range csv.Status.RequirementStatus {
+					result = append(result, v1alpha1.RequirementStatus{
+						Group:   s.Group,
+						Version: s.Version,
+						Kind:    s.Kind,
+						Name:    s.Name,
+						Status:  s.Status,
+					})
+				}
+				return result, nil
+			}).Should(ContainElement(
+				v1alpha1.RequirementStatus{
+					Group:   apiextensionsv1.SchemeGroupVersion.Group,
+					Version: apiextensionsv1.SchemeGroupVersion.Version,
+					Kind:    "CustomResourceDefinition",
+					Name:    crd.GetName(),
+					Status:  v1alpha1.RequirementStatusReasonPresent,
+				},
+			))
+		})
+
+		// Without this exception, upgrades can become blocked
+		// when the original CSV's CRD requirement becomes
+		// unsatisfied.
+		It("can satisfy an unassociated ClusterServiceVersion's ownership requirement if replaced by an associated ClusterServiceVersion", func() {
+			old := v1alpha1.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "unassociated-csv",
+					Namespace: testNamespace,
+				},
+				Spec: v1alpha1.ClusterServiceVersionSpec{
+					CustomResourceDefinitions: v1alpha1.CustomResourceDefinitions{
+						Owned: []v1alpha1.CRDDescription{{
+							Name:    "tests.example.com",
+							Version: "v1",
+							Kind:    "Test",
+						}},
+					},
+					InstallStrategy: newNginxInstallStrategy(genName("deployment-"), nil, nil),
+					InstallModes: []v1alpha1.InstallMode{
+						{
+							Type:      v1alpha1.InstallModeTypeAllNamespaces,
+							Supported: true,
+						},
+					},
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), &old)).To(Succeed())
+
+			new := v1alpha1.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "associated-csv",
+					Namespace: testNamespace,
+				},
+				Spec: v1alpha1.ClusterServiceVersionSpec{
+					Replaces: "unassociated-csv",
+					CustomResourceDefinitions: v1alpha1.CustomResourceDefinitions{
+						Owned: []v1alpha1.CRDDescription{{
+							Name:    "tests.example.com",
+							Version: "v1",
+							Kind:    "Test",
+						}},
+					},
+					InstallStrategy: newNginxInstallStrategy(genName("deployment-"), nil, nil),
+					InstallModes: []v1alpha1.InstallMode{
+						{
+							Type:      v1alpha1.InstallModeTypeAllNamespaces,
+							Supported: true,
+						},
+					},
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), &new)).To(Succeed())
+
+			Eventually(func() error {
+				return ctx.Ctx().Client().Get(context.Background(), client.ObjectKeyFromObject(&old), &old)
```

Comment:
@benluddy could you check my understanding?

So the transition path for unassociated looks like: created -> pending ("there's an annotation that's not mine on the CRD") -> associated created -> no longer pending ("I'm being replaced") -> deleted

Reply:
That's right. This spec should check that replacements do not block when the old CSV is not associated with its CRD, since that will be the state of most CSVs that pre-date this change.